npm-packages.md

```diff
@@ -1,14 +1,117 @@
-# Publishing and using `@guardian` NPM packages
+# NPM packages
 
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## Using
+## Publishing a `@guardian` library as an NPM package
 
--   Make sure you include any `@guardian` packages for transpilation in your build process
-    -   e.g. your [`webpack.config.js`](https://github.com/webpack/webpack/issues/2031#issuecomment-219040479) might including something that looks like this:
+### Authoring
+
+Write your library in [TypeScript](https://www.typescriptlang.org).
+
+### Packaging
+
+Do not include your TypeScript source code in your package. Instead, compile your source to JavaScript and [declaration files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) and publish those.
+
+> TypeScript depends on a project-specific configuration, and your setup may not be the same the project that installs your package
+
+Do not include or depend on [polyfills](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill). Let projects that install your package decide what they need (see the section on using our packages below).
+
+#### Compiling
+
+Use the TypeScript compiler (`tsc`) to generate the JavaScript and TypeScript [declaration files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) that you publish.
+
+Generate two versions of your library:
+
+1. ES modules targetting ES2020
+2. CommonJS modules targetting ES2018 (Node 10+)
```

Comment:
Out of interest, where did you find out what version of ECMAScript Node.js 10 supports?

Also maybe we should say Node 12+ here since we're about to kick everyone off of Node 10 😄

Reply:
i got it from SO :) but also https://node.green/#ES2018 

if we go with 12 we can target 2019

Reply:
Wow I have never scrolled down that far! Thanks for the tip! 🎩